lib/transaction/priority_queue.go

```diff
@@ -94,6 +97,13 @@ func NewPriorityQueue() *PriorityQueue {
 		pq:  make(priorityQueue, 0),
 		txs: make(map[common.Hash]*Item),
 	}
+
+	go metrics.CollectGaugeMetrics(
```

Comment:
same thing with this one. May make sense to create a `GaugeHarvester` that takes in a number of `GaugeCollector` implementations.